src/TwigComponent/src/Twig/ComponentNode.php

```diff
@@ -91,4 +91,20 @@ public function compile(Compiler $compiler): void
             ->raw("\n")
         ;
     }
+
+    /**
+     * Copied from Twig\Loader\FilesystemLoader, and adjusted to needs for this class
+     */
+    private function parseTemplateName(string  $name): mixed
+    {
+        if (isset($name[0]) && '@' == $name[0]) {
```

Comment:
```suggestion
        if ('@' === $name[0] ?? '') {
```
but also test is missing to prove the fix is working.

Reply:
Yea - a test would be ideal

Reply:
Test added, and branch rebased which now incorporates @weaverryan's latest test fixes. All green now here as well!

@stloyd I corrected the loose comparison. The (copied) code from Twig also used loose comparison. I guess because in this case loose comparison would be faster and if the type didn't match it would also fail (i.e. nothing would equal '@' anyway). For the same reason I kept the `isset` instead of `??` because it only needs one check to know to fail while the `??` would always do the assignment and then do the comparison always.